### ISSUE TITLE
Interactive markers should have an accessible name and `role="button"`

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#7428]
## Motivation

Interactive markers (**markers that bind to popups**) currently have no alternative text for screen readers (not announced, or announced as _blank_), they're also conveyed as images (not actionable buttons):

<img width="450" src="https://user-images.githubusercontent.com/26493779/105568669-7d34d480-5d3b-11eb-8a87-8d2375fe6de5.png" alt="inaccessible marker">

## Proposed solution

Interactive markers should have an accessible name by default, for example `alt="Marker"` (may be fixed by https://github.com/Leaflet/Leaflet/pull/7079/commits/f7e1cd9f5b5433ffb03784cfcdd5ba4b9955ca90 in https://github.com/Leaflet/Leaflet/pull/7079), and `role="button"`:

<img width="450" src="https://user-images.githubusercontent.com/26493779/105568661-5eced900-5d3b-11eb-9255-1ad8488a45f0.png" alt="accessible marker">

## Additional context
- MDN: [Interactive elements must be labeled](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Text_labels_and_names#interactive_elements_must_be_labeled)
- Related PRs: https://github.com/Leaflet/Leaflet/pull/7079, https://github.com/Leaflet/Leaflet/pull/6248.

**Comment by Malvoz** [comment]
It should be noted that due to the missing `role="button"` the map actually zooms in instead of opening the popup when using a screen reader (tested in NVDA) after pressing <kbd>Enter</kbd> when a marker is focused. This was a bit unexpected to me, other controls e.g. `<a>` that also are missing `role="button"` work fine in terms of keyboard operability, presumably because they're standard interactive elements.
